src/lib.rs

```diff
@@ -10,11 +10,65 @@ pub mod value;
 pub mod error;
 pub mod buffer;
 
-use internal::vm::{Module, Throw};
-use internal::mem::Handle;
-use internal::value::SomeObject;
+// The module version for Node.js 4.x is 46.
+// FIXME: detect this based on what we're compiling for.
+pub const NODE_MODULE_VERSION: i32 = 46;
 
-#[no_mangle]
-pub extern "C" fn neon_init(module: Handle<SomeObject>, init: fn(Module) -> Result<(), Throw>) {
-    Module::initialize(module, init);
+#[macro_export]
+macro_rules! onload {
+    ($module:ident, $init:block) => {
```

Comment:
No particular reason, just that it looks marginally better IMO.
Maybe `module => {...}`? At least it has some syntactical sense then.